### PR TITLE
ceph-dev-*: bump pbuilder dist when targeting Windows

### PR DESCRIPTION
**Pull request by petrutlucian94**
Ceph now uses c++ 20, thus requiring gcc 10 or later.

We'll need to update Windows jobs that leverage pbuilder to from Ubuntu Focal (20.04) to Ubuntu Jammy (22.04) to get mingw 8.0 and gcc 10.3.0.

Fixes: https://tracker.ceph.com/issues/58562

Signed-off-by: Lucian Petrut <lpetrut@cloudbasesolutions.com>

### CODE REVIEW

**Comment by idryomov** [comment]
> instead of Focal, which provides mingw 8.0 and gcc 10.3.0.

Did you mean to say "gcc 9.30" in the commit message?

File: ceph-dev-build/build/build_mingw [hunk 1]
```diff
@@ -1,8 +1,7 @@
 #!/bin/bash
 set -ex
 
-# We need Ubuntu Focal to cross-compile Ceph for Windows.
-# Previous versions provide broken Mingw packages.
+# We need Ubuntu jammy to cross-compile Ceph for Windows.
```
Comment (idryomov) [inline ceph-dev-build/build/build_mingw:4]: Nit: I think Jammy is to be capitalized just like Focal used to be ;)

**Comment by petrutlucian94** [comment]
> > instead of Focal, which provides mingw 8.0 and gcc 10.3.0.
> 
> Did you mean to say "gcc 9.3.0" here in the commit message?

nope, mingw 8.0 includes mingw-gcc 10.3.0:

```
$ dpkg -l | grep mingw

ii  gcc-mingw-w64                                               10.3.0-14ubuntu1+24.3                   all          
...
ii  mingw-w64                                                   8.0.0-1                                 all          Development environment targeting 32- and 64-bit Windows

$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.1 LTS
Release:        22.04
Codename:       jammy
```

File: ceph-dev-build/build/build_mingw (hunk 1 above)
Comment (petrutlucian94) [reply ceph-dev-build/build/build_mingw:4]: We'll eventually have to bump it again, using the same casing makes it easier to find/replace. If you prefer, I can update it.

**Comment by idryomov** [comment]
This is on Jammy but that part of the commit message talks about Focal.

**Comment by idryomov** [comment]
Perhaps reword this to say something like:

We'll need to update Windows jobs that leverage pbuilder to from Ubuntu Focal (20.04) to Ubuntu Jammy (22.04) to get mingw 8.0 and gcc 10.3.0.

**Comment by petrutlucian94** [comment]
Right, it was a bit confusing, thanks for the suggestion.

File: ceph-dev-build/build/build_mingw (hunk 1 above)
Comment (idryomov) [reply ceph-dev-build/build/build_mingw:4]: It looks like you updated both `setup_mingw` files but neither of `build_mingw` files.  This is really a nit and not blocking by any means but I think it's better to be consistent.  `git grep -i` works pretty well ;)

Comment (petrutlucian94) [reply ceph-dev-build/build/build_mingw:4]: Oh right, my bad. I'll send a new patch.

**Review by idryomov — APPROVED** [review]
